app/javascript/components/project-submissions/components/__tests__/submissions-list.test.jsx

```diff
@@ -0,0 +1,204 @@
+      expect(screen.queryAllByTestId('submission')[2].textContent).toBe('baz');
+    });
+
+    it('reorders submissions when a submission gains more likes than the one above it', () => {
+      // this test component is used to mock the user interaction of increasing the likes
+      // on a particular submission
+      const TestComponent = () => {
+        // setup dummy submissions to be able to change submission like numbers with an event
+        const [testSubmissions, setTestSubmissions] = useState(submissions);
+
+        // set 'baz' likes to 10, putting it first in the order instead of third
+        const reorderSubmissions = () => {
+          const newSubmissions = [...submissions];
+          const baz = newSubmissions.findIndex((sub) => sub.id === 'baz');
+          newSubmissions[baz] = { ...newSubmissions[baz], likes: 10 };
+
+          setTestSubmissions(newSubmissions);
+        };
+
+        return (
+          <>
+            <SubmissionsList
+              submissions={testSubmissions}
+              handleDelete={handleDelete}
+              handleUpdate={handleUpdate}
+              handleLikeToggle={handleLikeToggle}
+            />
+            <button
+              type="button"
+              onClick={reorderSubmissions}
+              data-test-id="reorder"
+            >
+              Click me
+            </button>
+          </>
+        );
+      };
+
+      render(
+        <ProjectSubmissionContext.Provider value={{ allSubmissionsPath: '#' }}>
+          <TestComponent />
+        </ProjectSubmissionContext.Provider>,
+      );
+
+      expect(screen.queryAllByTestId('submission')[0].textContent).toBe('foo');
+      expect(screen.queryAllByTestId('submission')[2].textContent).toBe('baz');
+
+      act(() => fireEvent.click(screen.getByTestId('reorder')));
+
+      expect(screen.queryAllByTestId('submission')[0].textContent).toBe('baz');
+      expect(screen.queryAllByTestId('submission')[1].textContent).toBe('foo');
+    });
+
+    it('does not render any submissions when array is empty', () => {
+      render(
+        <ProjectSubmissionContext.Provider value={{ allSubmissionsPath: '#' }}>
+          <SubmissionsList
+            submissions={[]}
+            handleDelete={handleDelete}
+            handleUpdate={handleUpdate}
+            handleLikeToggle={handleLikeToggle}
+          />
+        </ProjectSubmissionContext.Provider>,
+      );
+
+      expect(screen.queryAllByTestId('submission').length).toBe(0);
+    });
+
+    it('renders a no submissions yet message when array is empty', () => {
+      render(
+        <ProjectSubmissionContext.Provider value={{ allSubmissionsPath: '#' }}>
+          <SubmissionsList
+            submissions={[]}
+            handleDelete={handleDelete}
+            handleUpdate={handleUpdate}
+            handleLikeToggle={handleLikeToggle}
+          />
+        </ProjectSubmissionContext.Provider>,
+      );
+
+      expect(
```

Comment:
I think this can be reasonably included in the test above, since we are testing the same exact conditions. If we were looking to test more than 2 side-effects of a test condition, we could probably wrap them all in a new `describe` block. In this case I think combining them is fine.

Reply:
Good point! I didn't realize they were essentially testing the same thing till I read through them again.